identitykit/src/main/java/com/eldersoss/identitykit/oauth2/Token.kt

```diff
@@ -16,48 +16,35 @@
 
 package com.eldersoss.identitykit.oauth2
 
-import com.eldersoss.identitykit.Error
+import com.eldersoss.identitykit.errors.OAuth2InvalidTokenResponseError
 import com.eldersoss.identitykit.ext.getOptString
-import com.eldersoss.identitykit.getError
-import com.eldersoss.identitykit.network.NetworkResponse
+import org.json.JSONObject
 
 /**
  * @see <a href="https://tools.ietf.org/html/rfc6749#section-5.1">Successful Response</a>
  *  @constructor Access token object
  */
-data class Token(val accessToken: String,
-                 val tokenType: String,
-                 val expiresIn: Long,
-                 val refreshToken: String?,
-                 val scope: String?)
+class Token(private val jsonObject: JSONObject) {
+    val accessToken: String
+    val tokenType: String
+    val expiresIn: Long
+    val refreshToken: String?
+    val scope: String?
 
-/** Parse Token object from network response, possible error */
-fun parseToken(networkResponse: NetworkResponse, callback: (Token?, Error?) -> Unit) {
-    if (networkResponse.getJson() != null && networkResponse.statusCode in 200..299) {
-        val jsonObject = networkResponse.getJson()
-        jsonObject?.let {
-            try {
-                callback(
-                        Token(
-                                //required fields
-                                it.getString("access_token"),
-                                it.getString("token_type"),
-                                (System.currentTimeMillis() / 1000) + it.getLong("expires_in"),
-                                //optional fields
-                                it.getOptString("refresh_token"),
-                                it.getOptString("scope")
-                        ), null
-
-                )
-            } catch (e: Exception) {
-                callback(null, OAuth2Error.INVALID_TOKEN_RESPONSE)
-            }
+    init {
+        try {
+            accessToken = jsonObject.getString("access_token")
+            tokenType = jsonObject.getString("token_type")
+            expiresIn = (System.currentTimeMillis() / 1000) + jsonObject.getLong("expires_in")
```

Comment:
Store `expires_in` as provide `isExpired` property that internally calculates the expiration time